src/mlpack/methods/edge_boxes/feature_extraction.hpp

```diff
@@ -0,0 +1,99 @@
+/**
+ * @file feature_extraction.hpp
+ * @author Nilay Jain 
+ *
+ * Feature Extraction for the edge_boxes algorithm.
+ */
+#ifndef MLPACK_METHODS_EDGE_BOXES_STRUCTURED_TREE_HPP
+#define MLPACK_METHODS_EDGE_BOXES_STRUCTURED_TREE_HPP
+//#define INF 999999.9999
+//#define EPS 1E-20
+#include <mlpack/core.hpp>
+#include "feature_parameters.hpp"
+namespace mlpack {
+namespace structured_tree {
+
+template <typename MatType = arma::mat, typename CubeType = arma::cube>
+class StructuredForests
+{
+ private:
+  FeatureParameters params;
+
+ public:
+
+  static constexpr double eps = 1e-20;
```

Comment:
same as before, if there are no special, could you set it as private data member?

The lesser you let the users access you data directly, the more flexible you can refine your codes in the future without changing the api